~~~
-- The CUDA compiler identification is unknown
CMake Error at /vcpkg/downloads/tools/cmake-3.30.1-linux/cmake-3.30.1-linux-x86_64/share/cmake-3.30/Modules/CMakeDetermineCUDACompiler.cmake:266 (message):
  Failed to detect a default CUDA architecture.
~~~

Needs the extra CUDA scripting as recently added to port ggml and others.
https://github.com/microsoft/vcpkg/blob/master/ports/ggml/portfile.cmake#L35-L39

--- a/ports/magma/portfile.cmake
+++ b/ports/magma/portfile.cmake
@@ -24,14 +24,21 @@ vcpkg_extract_source_archive(
       fix-min-max.patch
 )
 
+
+
+vcpkg_find_cuda(OUT_CUDA_TOOLKIT_ROOT cuda_toolkit_root) 
+
 vcpkg_cmake_configure(
   SOURCE_PATH "${src_path}"
   OPTIONS
     -DMAGMA_ENABLE_CUDA=ON
     -DMAGMA_ENABLE_HIP=OFF # HIP is backend and seems additive?!
     -DUSE_FORTRAN=OFF
+    "-DCMAKE_CUDA_COMPILER:FILEPATH=${NVCC}"
+    "-DCUDAToolkit_ROOT=${cuda_toolkit_root}"
     ${opts}
 )
+
 
 vcpkg_cmake_install()
 

--- a/ports/magma/vcpkg.json
+++ b/ports/magma/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "magma",
   "version": "2.8.0",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Matrix Algebra on GPU and Multi-core Architectures (MAGMA) is a collection of next-generation linear algebra libraries for heterogeneous computing",
   "homepage": "https://icl.utk.edu/magma/",
   "license": null,

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6018,7 +6018,7 @@
     },
     "magma": {
       "baseline": "2.8.0",
-      "port-version": 1
+      "port-version": 2
     },
     "magnum": {
       "baseline": "2020.06",

--- a/versions/m-/magma.json
+++ b/versions/m-/magma.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "a54067bca358c87d6517a37b8360264e58097dc5",
+      "version": "2.8.0",
+      "port-version": 2
+    },
+    {
       "git-tree": "1f027fdb55e2bea18637a7cbe85c2ea544ee2f6d",
       "version": "2.8.0",
       "port-version": 1

